hamster_gtk/preferences/widgets/hamster_spin_button.py

```diff
@@ -43,32 +43,42 @@ class HamsterSpinButton(Gtk.SpinButton, ConfigWidget):
     # Required else you would need to specify the full module name in ui file
     __gtype_name__ = 'HamsterSpinButton'
 
-    def __init__(self, adj=None):
+    def __init__(self, adjustment=None, climb_rate=0, digits=0):
         """
         Initialize widget.
 
         Args:
-            adj (Gtk.Adjustment, SimpleAdjustment): Adjustment for the widget, either
+            adj (Gtk.Adjustment, SimpleAdjustment, optional): Adjustment for the widget, either
                 :class:`Gtk.Adjustment` or a :class:`SimpleAdjustment`.
-                See their respective documentation for more information.
+                See their respective documentation for more information. Defaults to ``None`` in
+                which case it can be set later.
+            climb_rate (float): See Gtk.SpinButton documentation.
+            digits (int): See Gtk.SpinButton documentation.
         """
-        super(Gtk.Entry, self).__init__()
-
-        if adj is not None:
-            if isinstance(adj, SimpleAdjustment):
-                if adj.min > adj.max:
-                    raise ValueError('Minimal value has to be lower than maximal value.')
-                if adj.step == 0:
-                    raise ValueError('Step value has to be non-zero.')
-
-                adjustment = Gtk.Adjustment(adj.min, adj.min, adj.max, adj.step, 10 * adj.step, 0)
-                self.configure(adjustment, adj.step, 0)
-                self.set_numeric(True)
-            elif isinstance(adj, Gtk.Adjustment):
-                self.set_adjustment(adj)
-            else:
+        super(Gtk.SpinButton, self).__init__()
+
+        self.set_numeric(True)
+
+        if adjustment is not None:
+            if not isinstance(adjustment, SimpleAdjustment) and not isinstance(adjustment,
```

Comment:
Maybe `isinstance(adjustment, (SimpleAdjustment, Gtk.Adjustment))` is more idiomatic? Not sure.